build/MSBuildToolset/build-toolset.ps1

```diff
@@ -1,24 +1,36 @@
 ﻿# Builds the toolset for use in unix msbuild-coreclr
 param (
-    [string]$NuGetTaskDir = "$PSScriptRoot\..\..\..\NuGet.BuildTasks\src\Microsoft.NuGet.Build.Tasks\bin\Debug"
+    [string]$NuGetTaskDir,
+    [string]$RoslynDropDir,
+    [string]$DotNetPath = "dotnet"
 )
 
+function ZipFiles($sourceDir, $zipPath)
```

Comment:
This is the cause of your line endings issue. The powershell zip functionality does not correctly encode line endings for Unix zip. Explorer is the easiest way to compatibly zip something on Windows.